node/src/components/consensus/era_supervisor.rs

```diff
@@ -635,23 +668,24 @@ where
     pub(super) fn handle_initialize_eras(
         &mut self,
         switch_blocks: HashMap<EraId, BlockHeader>,
-        validators: BTreeMap<PublicKey, U512>,
-        state_root_hash: Digest,
+        genesis_validators: BTreeMap<PublicKey, U512>,
+        genesis_state_root_hash: Digest,
```

Comment:
It's not _always_ about Genesis though?

Reply:
In this approach it is, actually. Any era other than era 0 is initialized based on the switch block, so these parameters are only used at genesis.

Reply:
Ah, and after we finally start reading the Genesis validators from the auction contract the approach will be unified?